myosuite/envs/myo/myochallenge/chasetag_v0.py

```diff
@@ -0,0 +1,376 @@
+        """
+        return np.array([self.sim.data.joint(joint).qpos for joint in self.opponent_joints], dtype=np.float32)[:,0].copy()
+
+    def _set_opponent_pose(self, pose):
+        """
+        Set opponent pose directly.
+        :param pose: Pose of the opponent.
+        :type pose: list -> [x, y, angle]
+        """
+        for name, val in zip(self.opponent_joints, pose):
+            self.sim.data.joint(name).qpos = val
+            self.sim.data.joint(name).qvel = 0.0
+
+    def _move_opponent(self, vel: list):
+        """
+        This is the main function that moves the opponent and should always be used if you want to physically move
+        it by giving it a velocity. If you want to teleport it to a new position, use `_set_opponent_pose`.
+        :param vel: Linear and rotational velocities in [-1, 1]. Moves opponent
+                  forwards or backwards and turns it. vel[0] is assumed to be linear vel and
+                  vel[1] is assumed to be rotational vel
+        :type vel: list -> [lin_vel, rot_vel].
+        """
+        self.opponent_vel = vel
+        assert len(vel) == 2
+        vel[0] = np.abs(vel[0])
+        vel = np.clip(vel, -1, 1)
+        pose = self._get_opponent_pose()
+        x_vel = vel[0] * np.cos(pose[-1]+0.5*np.pi)
+        y_vel = vel[0] * np.sin(pose[-1] +0.5*np.pi)
+        pose[0] -= self.dt * x_vel
+        pose[1] -= self.dt * y_vel
+        pose[2] += self.dt * vel[1] 
+        self._set_opponent_pose(pose)
+
+    def _random_movement(self):
+        """
+        This moves the opponent randomly in a correlated 
+        pattern.
+        """
+        return self.noise_process.sample()
+
+    def _sample_opponent_policy(self):
+        """
+        Takes in three probabilities and returns the policies with the given frequency.
+        """
+        rand_num = self.np_random.uniform()
+
+        if rand_num < self.opponent_probabilities[0]:
+            self.opponent_policy = 'static_stationary'
+        elif rand_num < self.opponent_probabilities[0] + self.opponent_probabilities[1]:
+            self.opponent_policy = 'stationary'
+        elif rand_num < self.opponent_probabilities[0] + self.opponent_probabilities[1] + self.opponent_probabilities[2]:
+            self.opponent_policy = 'random'
+
+    def _update_opponent_state(self):
+        """
+        This function executes an opponent step with 
+        one of the control policies.
+        """
+        if self.opponent_policy == 'stationary' or self.opponent_policy == 'static_stationary':
+            opponent_vel = np.zeros(2,)
+
+        elif self.opponent_policy == 'random':
+            opponent_vel = self._random_movement()
+
+        else:
+            raise NotImplementedError(f"This opponent policy doesn't exist. Chose: static_stationary, stationary or random. Policy was: {self.opponent_policy}")
+        self._move_opponent(opponent_vel)
+
+    def _initialize_opponent(self, qpos, qvel):
```

Comment:
Should the opponent be its own class? 
- it will make it easy for you to swap one with another by just importing the correct one. You can also specify that eval opponent will not be disclosed.